Cross-repo fails with `failed to load PR info`

There are occasions where the `enable_cross_pr_testing: true` setting on the `swift_pacakge_test.yml` fails with the following.


```
Run cat /__w/swift-package-manager/swift-package-manager/github-workflows/.github/workflows/scripts/cross-pr-checkout.swift > /tmp/cross-pr-checkout.swift
  cat /__w/swift-package-manager/swift-package-manager/github-workflows/.github/workflows/scripts/cross-pr-checkout.swift > /tmp/cross-pr-checkout.swift
  swift /tmp/cross-pr-checkout.swift "swiftlang/swift-package-manager" "10208"
  shell: sh -e {0}
Failed to load PR info from https://api.github.com/repos/swiftlang/swift-package-manager/pulls/10208: keyNotFound(CodingKeys(stringValue: "base", intValue: nil), Swift.DecodingError.Context(codingPath: [], debugDescription: "No value associated with key CodingKeys(stringValue: \"base\", intValue: nil) (\"base\").", underlyingError: nil))
Error: Process completed with exit code 1.
```

Here are some workflows runs:
- https://github.com/swiftlang/swift-package-manager/actions/runs/27564722105/job/81485694028?pr=10208

--- a/.github/workflows/scripts/cross-pr-checkout.swift
+++ b/.github/workflows/scripts/cross-pr-checkout.swift
@@ -117,11 +117,23 @@ func getPRInfo(repository: String, prNumber: String) async throws -> PRInfo {
     throw GenericError("Failed to form URL for GitHub API")
   }
 
-  do {
-    let data = try await downloadData(from: prInfoUrl)
-    return try JSONDecoder().decode(PRInfo.self, from: data)
-  } catch {
-    throw GenericError("Failed to load PR info from \(prInfoUrl): \(error)")
+  let maxAttempts = 5
+  var attempt = 1
+  while true {
+    do {
+      let data = try await downloadData(from: prInfoUrl)
+      return try JSONDecoder().decode(PRInfo.self, from: data)
+    } catch {
+      if attempt == maxAttempts {
+        throw GenericError("Failed to load PR info from \(prInfoUrl) after \(maxAttempts) attempts: \(error)")
+      }
+      let delaySeconds = UInt64(1 << (attempt - 1))
+      print(
+        "Failed to load PR info from \(prInfoUrl) (attempt \(attempt) of \(maxAttempts)): \(error). Retrying in \(delaySeconds)s..."
+      )
+      try await Task.sleep(nanoseconds: delaySeconds * 1_000_000_000)
+      attempt += 1
+    }
   }
 }
 

--- a/.github/workflows/scripts/cross-pr-checkout.swift
+++ b/.github/workflows/scripts/cross-pr-checkout.swift
@@ -131,7 +131,7 @@ func getPRInfo(repository: String, prNumber: String) async throws -> PRInfo {
       print(
         "Failed to load PR info from \(prInfoUrl) (attempt \(attempt) of \(maxAttempts)): \(error). Retrying in \(delaySeconds)s..."
       )
-      try await Task.sleep(nanoseconds: delaySeconds * 1_000_000_000)
+      try await Task.sleep(for: .seconds(delaySeconds))
       attempt += 1
     }
   }

